### ISSUE TITLE
Decide on naming conventions for suites of steps and their class names

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-tool_dataflows#145]
* [x] move all classes to local namespace
* [x] classes should either a prefix like 'source_xyz', 'connector, xyz', or nested ones like flow_filter_xyz
* [x] most of these rough groupings of steps should inherit from a generic class which sets the inputs and outputs
* [x] have methods which say what type or 'group' this step is which will be a language string used to group the steps in the step chooser page
* [ ] we might also need a sort order method for the groups. Both of these will be set in the classes we inherit

https://etherpad.catalyst.net.nz/dataflows-different-types-of-steps

**Comment by keevan** [comment]
Currently worked on in https://github.com/catalyst/moodle-tool_dataflows/pull/146

I've implemented most of the other points, but the 1st and last. The last one I didn't end up doing because I've rendered the ordering of options via a mustache template so it has a fixed order for now, but this can change if needed.

With the first point, what would the correct namespace be? 

For example, `namespace tool_dataflows\step;` is currently used for `class reader_sql extends reader_step`. Should this just nest under another local folder? `namespace tool_dataflows\local\step;` Reference (rule 1 2 and 3 @ https://docs.moodle.org/dev/Coding_style#Namespaces)
